habitat-baselines/habitat_baselines/rl/multi_agent/pop_play_wrappers.py

```diff
@@ -182,7 +182,9 @@ def act(
         agent_prev_actions = prev_actions.split(
             split_index_dict["index_len_prev_actions"], -1
         )
-        agent_masks = masks.split([1, 1], -1)
+        # TODO(ZXZ): modify here for four agents
+        # agent_masks = masks.split([1, 1], -1)
+        agent_masks = masks.split([1, 1, 1, 1], -1)
```

Comment:
Please initialize agent_masks with adaptive size, e.g.:
```suggestion
        agent_masks = masks.split([1] * n_agents, -1)
```

Reply:
Solved